dds/DCPS/transport/framework/TransportRegistry.cpp

```diff
@@ -621,11 +636,155 @@ bool TransportRegistry::has_type(const TransportType_rch& type) const
   return type_map_.count(name);
 }
 
-bool TransportRegistry::has_transport_template() const
+bool TransportRegistry::create_new_transport_instance_for_participant(DDS::DomainId_t id, OPENDDS_STRING& config_name)
 {
-  return transport_templates_.size() > 0;
+  TransportConfig_rch cfg = get_config(domain_config_map_[id]);
+
+  OPENDDS_STRING inst_name = cfg->instances_[0]->name() + "_" + config_name;
+  config_name = "transport_config_" + config_name;
+
+  OpenDDS::DCPS::TransportConfig_rch config =
+    TheTransportRegistry->create_config(config_name);
+  OpenDDS::DCPS::TransportInst_rch inst =
+    TheTransportRegistry->create_inst(inst_name, "rtps_udp");
+
+  ACE_Configuration_Heap ach;
+  ACE_Configuration_Section_Key sect_key;
+  ach.open();
+  ach.open_section(ach.root_section(), ACE_TEXT("the_transport_setup"), 1, sect_key);
+
+  if (TheServiceParticipant->belongs_to_domain_range(id)) {
+    TransportTemplate tr_inst;
+    get_transport_template_info(cfg->name().c_str(), tr_inst);
+
+    for (OPENDDS_MAP(OPENDDS_STRING, OPENDDS_STRING)::const_iterator it = tr_inst.transport_info.begin();
+         it != tr_inst.transport_info.end(); ++it) {
+      // customization.
+      OPENDDS_MAP(OPENDDS_STRING, OPENDDS_STRING)::const_iterator idx = tr_inst.customizations.find(it->first);
+      if (idx != tr_inst.customizations.end()) {
+        OPENDDS_STRING addr = it->second;
+
+        // only add_domain_id_to_ip_addr and add_domain_id_to_port are supported at this time.
+        if (idx->second.find("add_domain_id_to_ip_addr") != std::string::npos) {
```

Comment:
I know it's a pain, but I would prefer to see us parsing out the individual strings (looking for ",") so we can complain when we find something we don't expect. For example, if they misspell something, we can complain rather than just ignore it since we didn't find the exact string we were looking for.

As it stands, it looks like this will currently accept "thisiswrong_add_domain_id_to_port,totally_bogus" because it will find "add_domain_id_to_port".

Reply:
Yes, I agree. I'll make this change.